src/main/java/org/terasology/manualLabor/systems/ShearingSystem.java

```diff
@@ -6,59 +6,61 @@
-    public static final int HITS_TO_SHEAR = 5;
-    public static final ArrayList<ItemComponent> SHEARING_ITEMS = null; //TODO yet to decide
-    public static final int HAIR_REGROWTH_TIME = 3*60*1000; // In ms
+    public static final String SHEARING_ITEM = "ManualLabor:CrudeShears";
+    public static final int HAIR_REGROWTH_TIME = 3 * 60 * 1000; // 3 minutes in ms
     public static final String HAIR_REGROWTH_ACTION_ID = "ManualLabor:HairRegrowthAction";
     private static final Logger logger = LoggerFactory.getLogger(ShearingSystem.class);
 
     @In
     protected Time time;
 
     @In
-    private DelayManager delayManager;
+    private AssetManager assetManager;
 
     @In
-    private EntityManager entityManager;
-
-    @ReceiveEvent
-    public void onAnimalSpawn(AnimalSpawnEvent event, EntityRef entityRef) {
-        if (entityRef.getParentPrefab().getUrn().equals(new ResourceUrn("WildAnimals:Sheep"))) {
-            entityRef.addComponent(new ShearableComponent());
-        }
-    }
+    private DelayManager delayManager;
 
     @ReceiveEvent(components = {ShearableComponent.class})
     public void onAttack(AttackEvent event, EntityRef entityRef) {
         ShearableComponent component = entityRef.getComponent(ShearableComponent.class);
-        if (!component.sheared) {
-            int hits = component.hits;
-            if (hits < HITS_TO_SHEAR) {
-                component.hits++;
-            } else {
-                component.hits = 0;
+        EntityRef player = event.getInstigator();
+        CharacterHeldItemComponent characterHeldItemComponent = player.getComponent(CharacterHeldItemComponent.class);
+        EntityRef heldItem = characterHeldItemComponent.selectedItem;
+        Prefab parentPrefab = heldItem.getParentPrefab();
+        if (parentPrefab != null) {
+            if (!component.sheared && parentPrefab.getUrn().equals(new ResourceUrn(SHEARING_ITEM))) {
                 component.sheared = true;
                 component.lastShearingTimestamp = time.getGameTimeInMs();
-                delayManager.addPeriodicAction(entityRef, HAIR_REGROWTH_ACTION_ID, 0, HAIR_REGROWTH_TIME/20);
-                //TODO switch model
+                delayManager.addPeriodicAction(entityRef, HAIR_REGROWTH_ACTION_ID, 0, HAIR_REGROWTH_TIME / 20);
+                SkeletalMeshComponent skeletalMeshComponent = entityRef.getComponent(SkeletalMeshComponent.class);
+                Optional<SkeletalMesh> skeletalMesh = assetManager.getAsset("WildAnimals:shearedSheep",
+                        SkeletalMesh.class);
+                skeletalMesh.ifPresent(mesh -> skeletalMeshComponent.mesh = mesh);
+                Optional<Material> shearedMaterial = assetManager.getAsset("WildAnimals:shearedSheepSkin",
+                        Material.class);
+                shearedMaterial.ifPresent(texture -> skeletalMeshComponent.material = shearedMaterial.get());
+                entityRef.saveComponent(skeletalMeshComponent);
```

Comment:
I'm not so happy with this piece of code (and the one below for switching back).
First, this is kind of a code duplication, although admittedly the assets differ.
Second, I think it would be nice to have the asset uris as constants.
Third, `skeletalMeshComponent` might be `null` or either one of the optionals might be not present, but we don't catch that and either run into an NPE or end up with a mesh and a material that don't match.